test/versioning-strategies/prerelease.ts

```diff
@@ -0,0 +1,280 @@
+        scope: null,
+        bareMessage: 'some documentation',
+        notes: [],
+        references: [],
+        breaking: false,
+      },
+    ];
+    it('can bump a major', async () => {
+      const strategy = new PrereleaseVersioningStrategy();
+      const oldVersion = Version.parse('1.2.3');
+      const newVersion = await strategy.bump(oldVersion, commits);
+      expect(newVersion.toString()).to.equal('2.0.0');
+    });
+
+    it('can bump a major on pre major for breaking change', async () => {
+      const strategy = new PrereleaseVersioningStrategy();
+      const oldVersion = Version.parse('0.1.2');
+      const newVersion = await strategy.bump(oldVersion, commits);
+      expect(newVersion.toString()).to.equal('1.0.0');
+    });
+
+    it('can bump a minor pre major for breaking change', async () => {
+      const strategy = new PrereleaseVersioningStrategy({
+        bumpMinorPreMajor: true,
+      });
+      const oldVersion = Version.parse('0.1.2');
+      const newVersion = await strategy.bump(oldVersion, commits);
+      expect(newVersion.toString()).to.equal('0.2.0');
+    });
+  });
+
+  describe('with a feature', () => {
+    const commits = [
+      {
+        sha: 'sha1',
+        message: 'feat: some feature',
+        files: ['path1/file1.txt'],
+        type: 'feat',
+        scope: null,
+        bareMessage: 'some feature',
+        notes: [],
+        references: [],
+        breaking: false,
+      },
+      {
+        sha: 'sha2',
+        message: 'fix: some bugfix',
+        files: ['path1/file1.rb'],
+        type: 'fix',
+        scope: null,
+        bareMessage: 'some bugfix',
+        notes: [],
+        references: [],
+        breaking: false,
+      },
+      {
+        sha: 'sha3',
+        message: 'docs: some documentation',
+        files: ['path1/file1.java'],
+        type: 'docs',
+        scope: null,
+        bareMessage: 'some documentation',
+        notes: [],
+        references: [],
+        breaking: false,
+      },
+    ];
+    it('can bump a minor', async () => {
+      const strategy = new PrereleaseVersioningStrategy();
+      const oldVersion = Version.parse('1.2.3');
+      const newVersion = await strategy.bump(oldVersion, commits);
+      expect(newVersion.toString()).to.equal('1.3.0');
+    });
+    it('can bump a minor pre-major', async () => {
+      const strategy = new PrereleaseVersioningStrategy();
+      const oldVersion = Version.parse('0.1.2');
+      const newVersion = await strategy.bump(oldVersion, commits);
+      expect(newVersion.toString()).to.equal('0.2.0');
+    });
+    it('can bump a patch pre-major', async () => {
+      const strategy = new PrereleaseVersioningStrategy({
+        bumpPatchForMinorPreMajor: true,
+      });
+      const oldVersion = Version.parse('0.1.2');
+      const newVersion = await strategy.bump(oldVersion, commits);
+      expect(newVersion.toString()).to.equal('0.1.3');
+    });
+  });
+
+  describe('with a fix', () => {
```

Comment:
In this case:

- 1.0.0-beta01 => 1.0.0-beta02
- 2.0.0-beta01 => 2.0.0-beta02
- 1.0.1-beta01 => 1.0.1-beta02
- 1.1.0-beta01 => 1.1.0-beta02
- 1.1.1-beta01 => 1.1.1-beta01

Reply:
> * 1.1.1-beta01 => 1.1.1-beta01

This one should be `1.1.1-beta01 => 1.1.1-beta02`?

Reply:
Yes, sorry.